connect_ferc1_to_eia.py

```diff
@@ -42,497 +42,785 @@
-    logger.debug(f'train: {solver}: c-{c}, cw-{cw}, p-{p}, l1-{l1}')
-    lrc = rl.LogisticRegressionClassifier(solver=solver,
-                                          C=c,
-                                          class_weight=cw,
-                                          penalty=p,
-                                          l1_ratio=l1,
-                                          random_state=0,
-                                          multi_class=multi_class,
-                                          )
-    results, fscore, precision, accuracy = kfold_cross_val(
-        lrc=lrc,
-        n_splits=n_splits,
-        features_known=features_known,
-        known_index=training_index)
-    results_options = results_options.append(pd.DataFrame(
-        data={'solver': [solver],
-              'precision': [statistics.mean(precision)],
-              'f_score': [statistics.mean(fscore)],
-              'accuracy': [statistics.mean(accuracy)],
-              'c': [c],
-              'cw': [cw],
-              'penalty': [p],
-              'l1': [l1],
-              'multi_class': [multi_class],
-              'predictions': [len(results)],
-              # 'df': [results],
-              'coef': [lrc.coefficients],
-              'interc': [lrc.intercept],
-              },
-    ))
-    return results_options, lrc
-
-
-def test_model_parameters(features_known, training_index, n_splits):
-    """
-    Test and corss validate model parameters.
-
-    Args:
-        features_known (pandas.DataFrame)
-        training_index (pandas.MultiIndex)
-        n_splits (int)
-    Returns:
-        pandas.DataFrame: dataframe in which each record contains results of
-            each model run.
-
-    """
-    solvers = ['newton-cg', 'lbfgs', 'liblinear', 'sag', 'saga']
-    cs = [1, 10, 100, 1000]
-    cws = ['balanced', None]
-    ps = {'newton-cg': ['l2', 'none'],
-          'lbfgs': ['l2', 'none'],
-          'liblinear': ['l1', 'l2'],
-          'sag': ['l2', 'none'],
-          'saga': ['l1', 'l2', 'elasticnet', 'none'],
-          }
-    multi_classes = ['auto', 'ovr', 'multinomial']
-
-    results_options = pd.DataFrame()
-    for solver in solvers:
-        for c in cs:
-            for cw in cws:
-                for p in ps[solver]:
-                    if p == 'elasticnet':
-                        l1_ratios = [.1, .3, .5, .7, .9]
-                    else:
-                        l1_ratios = [None]
-                    for l1 in l1_ratios:
-                        # liblinear solver doesnt allow multinomial multi_class
-                        if solver == 'liblinear':
-                            multi_classes = ['auto', 'ovr']
+        return results_options
+
+    @staticmethod
+    def get_hyperparameters_options():
+        """
+        Generate a dictionary with sets of options for model hyperparameters.
+
+        Note: The looping over all of the hyperparameters options here feels..
```

Comment:
It seems to me that the data / dict / code should be structured such that the output dict is replaced by a function call.

Reply:
.. either that or maybe we should try and sort out Heap's algorithm.